source/js/components/sortEvents.js

```diff
@@ -15,121 +15,173 @@ function minutesSinceMidnight(time) {
-            let cont = false;
-            if (i === j) cont = true;
-            if (events[i].to <= events[j].from) cont = true;
-            if (events[i].from >= events[j].to) cont = true;
-
-            if (!cont) countIntersects.add(lane[j]);
+            if (i !== j && eventsIntercept(events[i], events[j]))
+                countIntersects.add(lane[j]);
         }
+
+        // delete the currant lane index
+        // bc just 1 event by itself would have
+        // no intersections
         countIntersects.delete(lane[i]);
-        res[i].width = 100 / (countIntersects.size + 1);
+
+        // store the denominator of the percentage of screen
+        // this event should take, to use later
+        res[i].width = countIntersects.size + 1;
     }
 
     // get the left of each event
-    // loop through each event
-    const possibleLeft = new Set();
+    // set them sequentially by end time
+    // try all positions and sees if it intersects
+    // with the events before it
+
+    const rects = []; // stores the running array of events that are placed down
+
     for (let i = 0; i < points.length; i += 1) {
         const index = points[i].ind;
 
-        if (points[i].from) {
-            // find first thing that doesn't exist in set
+        if (!points[i].from) {
+            // loop over all possible left positions
+            // to see which one doesn't intersect with
+            // previous events
             let count = 0;
-            while (possibleLeft.has(count) === true) count += 1;
-            possibleLeft.add(count);
+            let testRect = {};
+            const numInt = res[index].width;
+            const rectWidth = 100 / numInt;
+            for (; count < numInt; count += 1) {
+                // update the test rectangle that represents
+                // the collision box of this event
+                testRect = {
+                    from: events[index].from,
+                    to: events[index].to,
+                    left: count * rectWidth,
+                    right: (count + 1) * rectWidth,
+                };
+
+                // check if putting test rect
+                // intersect with any previous events
+                let intersect = false;
+                for (let j = 0; j < rects.length; j += 1) {
+                    if (rectsIntercept(testRect, rects[j])) {
+                        intersect = true;
+                        break;
+                    }
+                }
 
-            // set left
-            res[index].left = count * res[index].width;
+                if (!intersect) break;
+            }
 
-            // update vars
-            storeInd[index] = count;
-        } else {
-            // processing the end of this event
-            possibleLeft.delete(storeInd[index]);
+            // debug statements for testing
+            if (count === numInt) {
+                //console.log('nowhere to put event without intersection');
```

Comment:
Codacy has a fix for the issue: [Expected exception block, space or tab after '//' in comment.](https://app.codacy.com/gh/cse110-sp21-group31/cse110-sp21-group31/pullRequest?prid=7583982)
```suggestion
                // console.log('nowhere to put event without intersection');
```